Enhance notification rules documentation (#32870)
DataDog/documentation: content/en/monitors/notify/notification_rules.md

---

## Overview
Monitor notification rules are predefined sets of conditions that automate the process of alerting your team based on tags and rule logic. Instead of individually configuring recipients for every monitor, notification rules let you define once and automatically route all monitor notifications whose notification tagset matches the rule's scope.

<div class="alert alert-info">There is a default limit of 1000 rules per organization.</a>.</div>

## Setup
To create a Monitor Notification Rule in Datadog, navigate to the [**Monitors > Settings > Notification Rules**][1] page. Then, click the **New Rule** button in the upper right. 
{{< img src="/monitors/notifications/notification_rules/notification_rule_form-light.png" alt="Configuration for a notification rule showing scopes, routing conditions, recipients, and matching monitors" style="width:100%;" >}}

### Choose the scope

> Define which tags a monitor notification must have to be routed to this rule. Matching evaluates the notification tagset (the union of monitor tags and the firing group's tags for multi alert monitors). Monitor tags alone can satisfy the scope alone and that would still be considered a match. Only monitor and group tags participate in matching. Matching is case-insensitive.

Reviewer: ```suggestion
Define the required tags for a monitor notification to be routed to this rule. Matching evaluates the notification tagset (the union of monitor tags and the firing group's tags for multi alert monitors). Monitor tags alone can satisfy the scope, and this is still considered a match. Only monitor and group tags participate in matching. Matching is case-insensitive.
```

##### Examples

| Notification Rule scope | Explanation |
| ------------------- | ---------------------- |
| `service:web-store`       | Route any notification about the `web-store` service. |
| `service:web-store AND env:prod`       | Route any notification about the `web-store` service running on the `prod` environment. |
| `service:webstore AND  NOT env:staging`       | Route any notification about the `web-store` service that is **not** running on the `staging` environment. |
| `env:*`       | Route any notification that carry the `env:<value>` tag (either from monitor tags or group). |

For an example of this, see [Routing logic](#routing-logic).

#### Rule scope limitations
There are a few limitations that are **not supported** which include:

* Keyless tags, such as `prod AND service:(A or B)` or `prod`, aren't supported. Tags need to have a key, in this case for example `env:prod`.
* Partial wildcards (`service:web-*`) and question mark wildcards `service:auth?` are not supported. Wildcard is allowed only if used alone like `service:*`.
* Scope lenght up to 3000 characters.

### Configure the recipients

> Define the recipients to notify whenever a monitor notification matches the rule's scope. You can specify when and to whom a notification should be sent. Notifications can be sent to email or any integration channel. There is a limit of 50 notification recipients per rule. For more information, see [Notifications][2].

Reviewer: ```suggestion
Define the recipients to notify when a monitor notification matches the rule's scope. You can specify when and to which recipients a notification should be sent. Notifications can be sent to email or any integration channel. There is a limit of 50 notification recipients per rule. For more information, see [Notifications][2].
```

#### Conditional recipients
You can notify a specific recipient only when certain conditions are met. For example, route "alerts" to your on-call recipient, and send "warn" notifications to a Slack channel if they don't require immediate action.

### Name it

> Give the rule a clear name so it’s easy to recognize on the Manage page.

Reviewer: ```suggestion
Use a clear name so the rule is identifiable on the **Manage** page.
```

---

## Overview
Monitor notification rules are predefined sets of conditions that automate the process of alerting your team based on tags and rule logic. Instead of individually configuring recipients for every monitor, notification rules let you define once and automatically route all monitor notifications whose notification tagset matches the rule’s scope.

<div class="alert alert-info">There is a default limit of 1000 rules per organization.</a>.</div>

## Setup
To create a Monitor Notification Rule in Datadog, navigate to the [**Monitors > Settings > Notification Rules**][1] page. Then, click the **New Rule** button in the upper right. 
{{< img src="/monitors/notifications/notification_rules/notification_rules_form_with_conditional_recipients.png" alt="Configuration for a notification rule showing tag scopes, routing conditions, recipients, and matching monitors" style="width:100%;" >}}

### Choose the scope

> Define which tags a monitor notification must have to be routed to this rule. Matching evaluates the notification tagset - the union of monitor tags and the firing group’s tags (for multi-alert monitors). Monitor tags alone can satisfy the scope alone and that would still be considered a match. Only monitor and group tags participate in matching; matching is case-insensitive.

Reviewer: ```suggestion
Define which tags a monitor notification must have to be routed to this rule. Matching evaluates the notification tagset (the union of monitor tags and the firing group's tags for multi alert monitors). Monitor tags alone can satisfy the scope alone and that would still be considered a match. Only monitor and group tags participate in matching. Matching is case-insensitive.
```

- NOT (env:staging OR env:dev)

##### Wildcards
Only key:* is supported (for example, env:*). Partial wildcards like env:prod-* are not supported. key:* matches if that key exists anywhere in the notification tagset.

##### Multiple values for the same key
Use either env:(prod OR staging) or env:prod OR env:staging.

##### Quoting
Wrap values that contain spaces in quotes, for example: team:"data platform".

##### Examples

| Notification Rule scope | Explanation |
| ------------------- | ---------------------- |
| `service:web-store`       | Route any notification about the `web-store` service. |
| `service:web-store AND env:prod`       | Route any notification about the `web-store` service running on the `prod` environment. |
| `service:webstore AND  NOT env:staging`       | Route any notification about the `web-store` service that is **not** running on the `staging` environment. |
| `env:*`       | Route any notification that carry the `env:<value>` tag (either from monitor tags or group). |

For an example of this, see [Routing logic](#routing-logic).

#### Rule scope limitations
There are a few limitations that are **not supported** which include:

> * Keyless tags, such as `prod AND service:(A or B)` or just `prod`, aren't supported. Tags need to have a key, in this case for example `env:prod`.

Reviewer: ```suggestion
* Keyless tags, such as `prod AND service:(A or B)` or `prod`, aren't supported. Tags need to have a key, in this case for example `env:prod`.
```

Additionally, you can click the vertical three-dot menu on the notification rule to **Edit** or **Delete**.

### From an individual monitor
In your monitor configuration, the **Recipient Summary** shows recipients that are applied to the monitor by matching notification rules. On the **Monitor** edit page, you may also see rules that _could_ match when new groups report (multi alert monitors). The **Monitor** status page shows rules that match.

{{< img src="/monitors/notifications/notification_rules/monitor_matching_notification_rule.png" alt="Recipient summary field showing the notification recipients applied by notification rules" style="width:100%;" >}}

## How matching works

> - Notification tagset = monitor tags ∪ tags of the firing group (for multi alert monitors). If a key has multiple values across monitor/group, all values are considered.

Reviewer: what does "Notification tagset = monitor tags ∪ tags" mean? Typo?

Additionally, you can click the vertical three-dot menu on the notification rule to **Edit** or **Delete**.

### From an individual monitor
In your monitor configuration, the **Recipient Summary** shows recipients that are applied to the monitor by matching notification rules. On the **Monitor** edit page, you may also see rules that _could_ match when new groups report (multi alert monitors). The **Monitor** status page shows rules that match.

{{< img src="/monitors/notifications/notification_rules/monitor_matching_notification_rule.png" alt="Recipient summary field showing the notification recipients applied by notification rules" style="width:100%;" >}}

## How matching works

> - Notification tagset = monitor tags ∪ tags of the firing group (for multi alert monitors). If a key has multiple values across monitor/group, all values are considered.

Reviewer: perfect.

> - Notification tagset = monitor tags ∪ tags of the firing group (for multi-alert monitors). If a key has multiple values across monitor/group, all values are considered.

Reviewer: ```suggestion
- Notification tagset = monitor tags ∪ tags of the firing group (for multi alert monitors). If a key has multiple values across monitor/group, all values are considered.
```

Reviewer: What does `tags ∪ tags` mean? Typo?

- Currently matches: A rule matches if at least one reporting group, combined with monitor tags, satisfies the scope; or, if the monitor tags alone do. NOT is evaluated per candidate tagset, so a group with a denied value won’t match.

> - Could match when new groups report (multi-alert monitors, Monitor edit surface): Treat each group-by key as present with any value, constrained by the monitor query’s allow/deny filters.

Reviewer: ```suggestion
- Could match when new groups report (multi alert monitors, Monitor edit surface): Treat each group-by key as present with any value, constrained by the monitor query's allow/deny filters.
```
